Diplom2

Chaplin-86 opened the pull request:
Добавлены тесты на создание, логин и изменение данных пользователя;
Добавлены тесты на создание и получение заказов

Chaplin-86 committed:
"Добавлены классы User, UserClient, UserLogin, UserGenerator;
Написаны тесты на ручку создания пользователя (/api/auth/register)"

Chaplin-86 committed:
"Добавлен метод creds в классе UserLogin;
удалены комментарии в классе CreateUserTest;
написаны тесты на ручку авторизации (/api/auth/login)"

Chaplin-86 committed:
"Отредактированы методы в классе UserClient;
Добавлены тесты на обновление информации о пользователе на ручку /api/auth/user"

Chaplin-86 committed:
"Отредактированы методы в классе UserClient;
Добавлены тесты на обновление информации о пользователе на ручку /api/auth/user"

Chaplin-86 committed:
"В пакет Orders добавлены классы Ingredients, IngredientsClient, IngredientsData, OrderClient, OrderModel;
Добавлены тесты на создание заказов на ручку /api/orders"

Chaplin-86 committed:
"Добавлены тесты на получение заказов пользователя на ручку /api/orders"

Chaplin-86 committed:
added allure report

mkouznetsov-git commented on src/test/java/OrderTests/CreateOrderTests.java:
```diff
@@ -0,0 +1,194 @@
+    @Test
+    @DisplayName("Создание заказа с ингредиентами авторизованным пользователем")
+    @Description("POST/api/orders")
+    public void CreateAuthOrderWithIngredientsTest() {
```
Тест падает:
java.lang.AssertionError: 
Expected: <200>
     but: was <403>

mkouznetsov-git commented on src/test/java/OrderTests/GetOrderTests.java:
```diff
@@ -0,0 +1,75 @@
+    @Test
+    @DisplayName("Получение заказа авторизованного пользователя")
+    @Description("GET/api/orders")
+    public void GetAuthUserOrdersTest() {
```
Тест падает с ошибкой:
java.lang.AssertionError: 
Expected: <200>
     but: was <403>

mkouznetsov-git commented on src/test/java/UserTests/UpdateUserTests.java:
```diff
@@ -0,0 +1,186 @@
+import static org.junit.Assert.*;
+import static praktikum.UserGenerator.randomUser;
+
+public class UpdateUserTests {
```
Здесь все тесты падают с ошибкой:
java.lang.AssertionError: 
Expected: <200>
     but: was <403>

Chaplin-86 committed:
"Исправлен тест "Создание заказа с ингредиентами авторизованным пользователем" в классе CreateOrderTests;
Исправлен тест "Получение заказа авторизованного пользователя" в классе GetOrderTests;
Исправлены тесты в классе UpdateUserTests

mkouznetsov-git commented on src/test/java/UserTests/UpdateUserTests.java:
```diff
@@ -0,0 +1,186 @@
+        String newEmail = "Sirius_Black@yandex.ru";
+        notAuthUser.setEmail(newEmail);
+
+        ValidatableResponse updateNotAuthUserEmailResponse = userClient.updateUser(notAuthUser, accessToken.replace("Bearer ", ""));
```
Зачем передаёшь адекватный токен? Надо пустую строчку же передавать или какое-то невалидное значение, если хочешь словить ошибку авторизации. В других тестах тоже это надо поправить.

Chaplin-86 committed:
Исправлены тесты для неавторизованного пользователя в классе UpdateUserTests

mkouznetsov-git commented on src/test/java/UserTests/UpdateUserTests.java:
```diff
@@ -0,0 +1,183 @@
+
+        User createdUser = new User(user.getEmail(), user.getPassword(),user.getName());
+
+        ValidatableResponse loginResponse = userClient.loginUser(userLogin.creds(createdUser));
```
Метод creds статичный, поэтому к нему обращаться нужно по имени класса, а не объекта.

mkouznetsov-git commented on src/test/java/UserTests/UpdateUserTests.java:
```diff
@@ -0,0 +1,183 @@
+
+        User createdUser = new User(user.getEmail(), user.getPassword(),user.getName());
+
+        ValidatableResponse loginResponse = userClient.loginUser(userLogin.creds(createdUser));
```
То же самое

mkouznetsov-git commented on src/test/java/UserTests/UpdateUserTests.java:
```diff
@@ -0,0 +1,183 @@
+
+        User createdUser = new User(user.getEmail(), user.getPassword(),user.getName());
+
+        ValidatableResponse loginResponse = userClient.loginUser(userLogin.creds(createdUser));
```
То же самое

mkouznetsov-git commented on src/test/java/UserTests/LoginUserTests.java:
```diff
@@ -0,0 +1,106 @@
+
+        userClient.createUser(user);
+
+        ValidatableResponse responseToLogin = userClient.loginUser(userLogin.creds(user));
```
Статичный метод

mkouznetsov-git commented on src/test/java/UserTests/LoginUserTests.java:
```diff
@@ -0,0 +1,106 @@
+
+        user.setEmail("harry@yandex.ru");
+
+        ValidatableResponse responseToLogin = userClient.loginUser(userLogin.creds(user));
```
Статичный метод

mkouznetsov-git commented on src/test/java/UserTests/LoginUserTests.java:
```diff
@@ -0,0 +1,106 @@
+
+        user.setPassword("pass");
+
+        ValidatableResponse responseToLogin = userClient.loginUser(userLogin.creds(user));
```
Статичный метод

Chaplin-86 committed:
Исправлено обращение к методу "creds" в классах LoginUserTests и UpdateUserTests

mkouznetsov-git commented on src/test/java/UserTests/LoginUserTests.java:
```diff
@@ -0,0 +1,106 @@
+    private static final String BASE_URL = "https://stellarburgers.nomoreparties.site";
+    private User user;
+    private UserClient userClient;
+    private UserLogin userLogin;
```
Перемення теперь не нужна. Удаляй.

mkouznetsov-git commented on src/test/java/UserTests/UpdateUserTests.java:
```diff
@@ -0,0 +1,183 @@
+    private static final String BASE_URL = "https://stellarburgers.nomoreparties.site";
+    private User user;
+    private UserClient userClient;
+    private UserLogin userLogin;
```
Перемення теперь не нужна. Удаляй.

Chaplin-86 committed:
Удалена лишняя переменная 'private UserLogin userLogin' в классах LoginUserTests и UpdateUserTests

Chaplin-86 committed:
Изменила передачу токена в методе удаления пользователя
